src/main/kotlin/dev/minn/jda/ktx/JDABuilder.kt

```diff
@@ -0,0 +1,399 @@
+    builder: InlineJDABuilder.() -> Unit
+                   ) =
+    InlineJDABuilder(JDABuilder.createLight(token, intent, *intents)).also(builder).build()
+
+inline fun LightJDA(
+    token: String? = null,
+    intents: Collection<GatewayIntent?>,
+    builder: InlineJDABuilder.() -> Unit
+                   ) = InlineJDABuilder(JDABuilder.createLight(token, intents)).also(builder).build()
+
+inline fun JDA(
+    token: String? = null,
+    intent: GatewayIntent,
+    vararg intents: GatewayIntent,
+    builder: InlineJDABuilder.() -> Unit
+              ) = InlineJDABuilder(JDABuilder.create(token, intent, *intents)).also(builder).build()
+
+inline fun JDA(
+    token: String? = null,
+    intents: Collection<GatewayIntent?>,
+    builder: InlineJDABuilder.() -> Unit
+              ) = InlineJDABuilder(JDABuilder.createLight(token, intents)).also(builder).build()
+
+inline fun DefaultJDABuilder(
+    token: String? = null,
+    builder: InlineJDABuilder.() -> Unit
+                            ) =
+    InlineJDABuilder(JDABuilder.createDefault(token)).also(builder)
+
+inline fun DefaultJDABuilder(
+    token: String? = null,
+    intent: GatewayIntent,
+    vararg intents: GatewayIntent,
+    builder: InlineJDABuilder.() -> Unit
+                            ) =
+    InlineJDABuilder(JDABuilder.createDefault(token, intent, *intents)).also(builder)
+
+inline fun DefaultJDABuilder(
+    token: String? = null,
+    intents: Collection<GatewayIntent?>,
+    builder: InlineJDABuilder.() -> Unit
+                            ) =
+    InlineJDABuilder(JDABuilder.createDefault(token, intents)).also(builder)
+
+inline fun LightJDABuilder(
+    token: String? = null,
+    builder: InlineJDABuilder.() -> Unit
+                          ) =
+    InlineJDABuilder(JDABuilder.createLight(token)).also(builder)
+
+inline fun LightJDABuilder(
+    token: String? = null,
+    intent: GatewayIntent,
+    vararg intents: GatewayIntent,
+    builder: InlineJDABuilder.() -> Unit
+                          ) =
+    InlineJDABuilder(JDABuilder.createLight(token, intent, *intents)).also(builder)
+
+inline fun LightJDABuilder(
+    token: String? = null,
+    intents: Collection<GatewayIntent?>,
+    builder: InlineJDABuilder.() -> Unit
+                          ) =
+    InlineJDABuilder(JDABuilder.createLight(token, intents)).also(builder)
+
+inline fun JDABuilder(
+    token: String? = null,
+    intent: GatewayIntent,
+    vararg intents: GatewayIntent,
+    builder: InlineJDABuilder.() -> Unit
+                     ) =
+    InlineJDABuilder(JDABuilder.create(token, intent, *intents)).also(builder)
+
+inline fun JDABuilder(
+    token: String? = null,
+    intents: Collection<GatewayIntent?>,
+    builder: InlineJDABuilder.() -> Unit
+                     ) =
+    InlineJDABuilder(JDABuilder.createLight(token, intents)).also(builder)
+
+class InlineJDABuilder(val builder: JDABuilder) {
+    var gatewayEncoding: GatewayEncoding? = null
```

Comment:
Shouldn't these defaults properly reflect the state of the builder?

Reply:
Perhaps they should, but they can also change at any time. But when they're set to null, it preserves the default value.